benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/AzureMatcherBenchmark.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    // Generated from https://github.com/Azure/azure-rest-api-specs
+    public partial class AzureMatcherBenchmark : MatcherBenchmarkBase
+    {
+        private const int SampleCount = 100;
+
+        private Matcher _route;
+        private Matcher _tree;
+
+        private int[] _samples;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            SetupEndpoints();
+
+            SetupRequests();
+
+            // The perf is kinda slow for these benchmarks, so we do some sampling
+            // of the request data.
+            _samples = SampleRequests(EndpointCount, SampleCount);
+
+            _route = SetupMatcher(RouteMatcher.CreateBuilder());
+            _tree = SetupMatcher(TreeRouterMatcher.CreateBuilder());
+        }
+
+        [Benchmark(OperationsPerInvoke = SampleCount)]
+        public async Task LegacyRoute()
+        {
+            var feature = new EndpointFeature();
```

Comment:
What do you think of moving allocating EndpointFeature out of benchmarks? Could have a single instance on a field and reuse.